prbt_hardware_support/test/integrationtests/integrationtest_speed_observer_and_operation_mode.cpp

```diff
@@ -0,0 +1,290 @@
+static const std::string BARRIER_NO_SVC_SUCESS{ "BARRIER_NO_SVC_SUCESS" };
+
+static const double TEST_FREQUENCY{ 10 };
+
+MATCHER_P(StoState, x, "Sto state " + std::string(negation ? "is not" : "is") + ": " + PrintToString(x) + ".")
+{
+  return arg.data == x;
+}
+
+MATCHER_P(ContainsAllNames, names,
+          "Names" + PrintToString(names) + std::string(negation ? "are not" : "are") + " in message.")
+{
+  for (auto& n : names)
+  {
+    bool found{ false };
+    for (auto& argn : arg->name)
+      if (argn.compare(n) == 0)
+        found = true;
+    if (!found)
+      return false;
+  }
+
+  return true;
+}
+
+class SpeedObserverIntegrationTest : public testing::Test, public testing::AsyncTest
+{
+public:
+  void SetUp() override;
+  void TearDown() override;
+
+  MOCK_METHOD2(operation_mode_cb, bool(GetOperationMode::Request& req, GetOperationMode::Response& res));
+  MOCK_METHOD2(sto_cb, bool(std_srvs::SetBool::Request& req, std_srvs::SetBool::Response& res));
+  MOCK_METHOD1(frame_speeds_cb, void(const FrameSpeeds::ConstPtr& msg));
+
+  void publishJointStatesAtSpeed(double v);
+  void stopJointStatePublisher();
+  void publishOm(OperationModes::_value_type omv);
+
+protected:
+  ros::NodeHandle nh_;
+  ros::NodeHandle pnh_{ "~" };
+  ros::Subscriber subscriber_;
+  ros::ServiceServer operation_mode_srv_;
+  ros::ServiceServer sto_srv_;
+  ros::Publisher fake_controller_joint_states_pub_;
+  ros::Publisher operation_mode_pub_;
+  std::vector<std::string> additional_frames_;
+  bool joint_publisher_running_{ false };
+  std_srvs::SetBool::Response sto_res;
+};
+
+void SpeedObserverIntegrationTest::SetUp()
+{
+  subscriber_ =
+      nh_.subscribe<FrameSpeeds>(FRAME_SPEEDS_TOPIC_NAME, 1, &SpeedObserverIntegrationTest::frame_speeds_cb, this);
+  fake_controller_joint_states_pub_ =
+      nh_.advertise<sensor_msgs::JointState>(FAKE_CONTROLLER_JOINT_STATES_TOPIC_NAME, 1);
+  operation_mode_pub_ =
+      nh_.advertise<OperationModes>(OPERATION_MODE_TOPIC, 10);
+
+  pnh_.getParam(ADDITIONAL_FRAMES_PARAM_NAME, additional_frames_);
+  ROS_DEBUG_STREAM("SetUp pnh:" << pnh_.getNamespace() << " nh:" << nh_.getNamespace());
+  for (auto& f : additional_frames_)
+  {
+    ROS_DEBUG_STREAM("- " << f);
+  }
+
+  operation_mode_srv_ =
+      nh_.advertiseService(OPERATION_MODE_SERVICE, &SpeedObserverIntegrationTest::operation_mode_cb, this);
+  sto_srv_ = nh_.advertiseService(STO_SERVICE, &SpeedObserverIntegrationTest::sto_cb, this);
+
+  sto_res.success = true;
+  sto_res.message = "testing ...";
+}
+
+void SpeedObserverIntegrationTest::TearDown()
+{
+  joint_publisher_running_ = false;
+  nh_.shutdown();
+}
+
+void SpeedObserverIntegrationTest::publishOm(OperationModes::_value_type omv){
+  OperationModes om;
+  om.value = omv;
+  om.time_stamp = ros::Time::now();
+  operation_mode_pub_.publish(om);
+}
+
+void SpeedObserverIntegrationTest::publishJointStatesAtSpeed(double v)
```

Comment:
```suggestion
void SpeedObserverIntegrationTest::publishJointStatesAtSpeed(double vel)
```

Reply:
really?

Reply:
is vel better than v? Why not velocity, then?